external-crates/move/crates/move-compiler/src/unit_test/plan_builder.rs

```diff
@@ -161,17 +178,32 @@ fn build_test_info<'func>(
         ))
     }
 
-    let test_annotation_params = parse_test_attribute(context, test_attribute, 0);
+    let test_annotation_params = parse_test_attribute(context, test_attribute, is_rand_test, 0);
     let mut arguments = Vec::new();
-    for (_mut, var, _) in &function.signature.parameters {
+    for (_mut, var, s_type) in &function.signature.parameters {
         let sp!(vloc, var_) = var.0;
         let var_ = match display_var(var_) {
             crate::hlir::translate::DisplayVar::Orig(s) => s.into(),
             crate::hlir::translate::DisplayVar::MatchTmp(_) => panic!("ICE temp as parameter"),
             crate::hlir::translate::DisplayVar::Tmp => panic!("ICE temp as parameter"),
         };
         match test_annotation_params.get(&var_) {
-            Some(value) => arguments.push(value.clone()),
+            Some(value) => arguments.push(TestArgument::Value(value.clone())),
+            None if is_rand_test => {
+                let generated_type = match convert_builtin_type_to_typetag(&s_type.value) {
```

Comment:
Yea, totally agreed -- definitely want to extend the number of supported types here over time :) And/or if we could auto-generate more complex datatypes on the fly since we have the module available to us so should be able to generate the layout and everything.